zeppelin-interpreter/src/main/java/org/apache/zeppelin/interpreter/remote/AppendOutputRunner.java

```diff
@@ -0,0 +1,112 @@
+package org.apache.zeppelin.interpreter.remote;
+
+import java.util.HashMap;
+import java.util.LinkedList;
+import java.util.List;
+import java.util.Map;
+import java.util.concurrent.BlockingQueue;
+import java.util.concurrent.LinkedBlockingQueue;
+
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+/**
+ * This thread sends paragraph's append-data
+ * periodically, rather than continously, with
+ * a period of BUFFER_TIME_MS. It handles append-data
+ * for all paragraphs across all notebooks.
+ */
+public class AppendOutputRunner implements Runnable {
+
+  private static final Logger logger =
+      LoggerFactory.getLogger(AppendOutputRunner.class);
+  private static RemoteInterpreterProcessListener listener;
+
+  private static final Long BUFFER_TIME_MS = new Long(100);
+  private static final Long SAFE_PROCESSING_TIME = new Long(10);
+  private static final Long SAFE_PROCESSING_STRING_SIZE = new Long(100000);
+
+  private static final BlockingQueue<AppendOutputBuffer> QUEUE =
+      new LinkedBlockingQueue<AppendOutputBuffer>();
+
+  @Override
+  public void run() {
+
+    while (true) {
```

Comment:
It looks like you are not familiar with `ScheduledExecutorService`. If you would write codes like this, only one executor will run forever. Can you check an example of `ScheduledExecutorService`? You will know this while loop is redundant.